source/Xunit.Gherkin.Quick/CoreModel/ScenarioExecutor.cs

```diff
@@ -24,13 +24,15 @@ public async Task ExecuteScenarioAsync(Feature featureInstance, string scenarioN
             var featureClass = FeatureClass.FromFeatureInstance(featureInstance);
             var featureFile = _featureFileRepository.GetByFilePath(featureClass.FeatureFilePath);
 
-			var gherkinScenario = featureFile.GetScenario(scenarioName);
+            var gherkinBackground = featureFile.GetBackground();
+            var gherkinScenario = featureFile
+                .GetScenario(scenarioName)
+                ?.ApplyBackground(gherkinBackground);
```

Comment:
This makes me think that ApplyBackground accepts null too.
Let's disallow that.
So here, we should pass in the background only if it's not null.
That kind of strict design will ensure that there is no unpredictable of double-intent code. Helps a great deal with edge cases and maintaining.